seglearn/transform.py

```diff
@@ -1422,3 +1424,152 @@ def transform(self, X):
+        '''
+        obj = _InitializePickableSampler()
+        obj.__class__ = patch_sampler(sampler_class)
+        return obj
+
+
+def patch_sampler(sampler_class):
+    '''
+    Return a dynamically patched imbalanced-learn Sampler class compatible with Pype.
+    '''
+    conditions = [
+        hasattr(sampler_class, 'fit_resample'),
+        hasattr(sampler_class, '_check_X_y'),
+        hasattr(sampler_class, '_get_param_names'),
+    ]
+    if not all(conditions):
+        raise TypeError('The sampler class to be patched must have a "fit_resample", a "_check_X_y"'
+                        ' method and a "_get_param_names" class method.')
+
+    class PickableSampler(sampler_class, XyTransformerMixin):
+        '''
+        Dynamically created (pickable) class derived from an imbalanced-learn Sampler and the
+        XyTransformerMixin in order to enable the use of the imbalanced-learn Sampler transforms
+        inside a seglearn Pype.
+
+        Parameters
+        ----------
+        shuffle : boolean, optional (default=False)
+        random_state : int, RandomState instance or None, optional (default=None)
+            seed of the pseudo random number generator used for shuffling the resampled data
+        **kwargs : keyword arguments to be passed to the imbalanced-learn Sampler base class
+
+        Returns
+        -------
+        self : object
+            returns self
+        '''
+        def __init__(self, shuffle=False, random_state=None, **kwargs):
+            # set shuffle and random_state
+            self.shuffle = shuffle
+            self.random_state = random_state
+
+            # call imbalanced-learn Sampler base class with the correct arguments
+            orig_signature = signature(super(PickableSampler, self).__init__)
+            orig_args = [p.name for p in orig_signature.parameters.values()
+                         if p.name != 'self' and p.kind != p.VAR_KEYWORD]
+            if "shuffle" in orig_args:
+                kwargs["shuffle"] = shuffle
+            if "random_state" in orig_args:
+                kwargs["random_state"] = random_state
+            super(PickableSampler, self).__init__(**kwargs)
+
+        @classmethod
+        def _get_param_names(cls):
+            '''
+            Get parameters of the imbalanced-learn Sampler base class and the additional arguments
+            of the dynamically derived class.
+            '''
+            init_signature = signature(getattr(cls, '__init__'))
+            parameters = [p.name for p in init_signature.parameters.values()
+                          if p.name != 'self' and p.kind != p.VAR_KEYWORD]
+            return sorted(set(sampler_class._get_param_names() + parameters))
+
+        @staticmethod
+        def _check_X_y(Xt, yt):
+            '''
+            Circumvent the check whether dim(Xt) == 2.
+            '''
+            Xt_2d = Xt.reshape(Xt.shape[0], -1)
```

Comment:
This creates a copy of Xt when Fortran-like (default) ordering is used. #24 solves this issue by choosing C-like ordering for the segmentation. Shall I give a warning if Fortran-like ordering is used, or shall I remove the "faked" check altogether?